lib/spack/spack/ci.py

```diff
@@ -805,13 +823,15 @@ def generate_gitlab_ci_yaml(
 
     temp_storage_url_prefix = None
     if "temporary-storage-url-prefix" in ci_config:
+        tty.warn("Support for temporary-storage-url-prefix will be removed in Spack 0.23")
         temp_storage_url_prefix = ci_config["temporary-storage-url-prefix"]
 
     # If a remote mirror override (alternate buildcache destination) was
     # specified, add it here in case it has already built hashes we might
     # generate.
+    # TODO: Remove this block in Spack 0.23
     mirrors_to_check = None
-    if remote_mirror_override:
+    if deprecated_mirror_config and remote_mirror_override:
```

Comment:
I am not sure if it makes sense to disable this based on the `deprecated_mirror_config` flag since it is still a CLI option.

Reply:
I don't want to modify the mirror state unless the user didn't use the new approach of adding a mirror named `buildcache-destination` and instead used that command line option.  Did I get that logic wrong?

Reply:
The option `--buildcache-destination` can still be specified without error, even in the new method. I think either this should be an error earlier saying we don't support both, or this specific override method should be allowed even in the new mode.

I didn't comment on the other mirror gymnastics we were doing because those were sort of magic from the outside world perspective.

Reply:
Ok, I'm leaning toward an error saying we don't support both at the same time, unless you object.

Reply:
That is also my inclination, having both active at the same time feels odd.